ruby/src/main/java/com/google/protobuf/jruby/RubyMap.java

```diff
@@ -189,8 +190,25 @@ public IRubyObject index(ThreadContext context, IRubyObject key) {
      */
     @JRubyMethod(name = "==")
     public IRubyObject eq(ThreadContext context, IRubyObject _other) {
-        if (_other instanceof RubyHash)
-            return toHash(context).op_equal(context, _other);
+        if (_other instanceof RubyHash){
+            // quick check to see if they are equivalent
+            IRubyObject matched = toHash(context).op_equal(context, _other);
+            if (matched == context.runtime.getTrue()) {
+                return context.runtime.getTrue();
+            }
+
+            RubyHash other = (RubyHash)_other;
```

Comment:
@cfallin another thought; the original logic would allow equivalency with a hash.  I'm not sure that is really the correct behavior.  So if we change it to where if the incoming obj is not equivalent to this, return false.